modin/pandas/test/dataframe/test_default.py

```diff
@@ -472,13 +484,33 @@ def test_mad_level(level):
     "value_vars", [lambda df: df.columns[-1], lambda df: df.columns[-4:], None]
 )
 def test_melt(data, id_vars, value_vars):
+    if StorageFormat.get() == "Hdk":
+        # Drop NA and sort by all columns to make sure the order
+        # is identical to Pandas.
+        def melt(df, *args, **kwargs):
+            df = df.melt(*args, **kwargs).dropna()
+            return df.sort_values(df.columns.tolist())
+
+        # HDK has different values of category codes than Pandas.
+        # The default comparator fails for categorical dtypes.
+        def comparator(df1, df2):
+            assert all(df1.index == df2.index)
+            for c1, c2 in zip(df1.columns, df2.columns):
+                assert c1 == c2
+                assert all(df1[c1] == df2[c2])
```

Comment:
can we just add the respective flag to `df_equals`?
```suggestion
            df_equals(df1, df2, ignore_categories=True)
```

also, I think that we can simplify the actual logic of comparison as follows:
```python
df1_cat_cols = df1.select_dtypes(include="category")
df2_cat_cols = df2.select_dtypes(include="category")
assert df1_cat_cols.equals(df1_cat_cols)
df1 = df1.astype({key: "string" for key in df1_cat_cols})
df2 = df2.astype({key: "string" for key in df2_cat_cols})
# ... continue with regular comparison
```

Reply:
Seems this code is not required any more. Removed.